python/lsst/ts/mtdomecom/mtdome_com.py

```diff
@@ -648,7 +676,17 @@ async def write_then_read_reply(
             if command not in disabled_commands:
                 self.log.debug(f"Sending {command_dict=}.")
                 await self.client.write_json(data=command_dict)
-                data = await asyncio.wait_for(self.client.read_json(), timeout=_TIMEOUT)
+                try:
+                    data = await asyncio.wait_for(
+                        self.client.read_json(), timeout=_TIMEOUT
+                    )
+                except TimeoutError as exc:
+                    self.communication_error_report = {
+                        "command_name": CommandName(command_name),
+                        "exception": exc,
+                        "response_code": ResponseCode.UNSUPPORTED,
+                    }
+                    raise exc
```

Comment:
This means CSC and GUI still need to deal with this hidden raised Error... Otherwise, they will be stuck when this error happens. This should be documented in the doc string somewhere...